cbor2/types.py

```diff
@@ -46,6 +46,15 @@ def __repr__(self):
         return 'CBORSimpleValue({self.value})'.format(self=self)
 
 
+class HashableMap(dict):
```

Comment:
No, just no. This is fundamentally wrong. This is still a mutable type, you're just trying to work around the fact that dicts are not hashable, as they rightfully should not be. Another type could be made where the keys and values cannot be changed but this is not it.

Reply:
Have a look at my proposal below